icu4c/source/i18n/dtfmtsym.cpp

```diff
@@ -1782,8 +1803,43 @@ struct CalendarDataSink : public ResourceSink {
                 arraySizes.puti(path, dataArraySize, errorCode);
                 if (U_FAILURE(errorCode)) { return; }
             } else if (value.getType() == URES_TABLE) {
-                // We are not on a leaf, recursively process the subtable.
-                processResource(path, key, value, errorCode);
+                // We might have an eras table that is replacing an eras leaf array
+                if (path.startsWith(kErasTagUChar, UPRV_LENGTHOF(kErasTagUChar))) {
+                    // path is one of eras/wide, eras/abbreviated, eras/narrow
+                    ResourceTable rDataTable = value.getTable(errorCode);
+                    int32_t dataTableSize = rDataTable.getSize();
+                    LocalArray<EraCodeAndName> eraCodeAndNames(new EraCodeAndName[dataTableSize], errorCode);
+                    // Collect the era entries that we do have, then sort by era code
```

Comment:
This seems roundabout... please take a look at my "suggestion 2" on the Java code. UVector should work in lieu of Java ArrayList.

Reply:
Yes, working on this now